src/usd/sdf_usd_parser/geometry.cc

```diff
@@ -0,0 +1,296 @@
+    // To keep things simple for now, a plane will only be parsed if its normal is
+    // the unit z vector (0, 0, 1).
+    // TODO(adlarkin) support plane normals other than the unit z vector (can update
+    // comment above once this functionality is added)
+    // TODO(adlarkin) update this to use the pxr::USDGeomPlane class when it's added
+    if (sdfPlane->Normal() != ignition::math::Vector3d::UnitZ)
+    {
+      std::cerr << "Only planes with a unit z vector normal are supported\n";
+      return false;
+    }
+
+    sdf::Box box;
+    box.SetSize(ignition::math::Vector3d(
+          sdfPlane->Size().X(), sdfPlane->Size().Y(), usd::kPlaneThickness));
+
+    sdf::Geometry planeBoxGeometry;
+    planeBoxGeometry.SetType(sdf::GeometryType::BOX);
+    planeBoxGeometry.SetBoxShape(box);
+
+    return ParseSdfBoxGeometry(planeBoxGeometry, _stage, _path);
+  }
+
+  bool ParseSdfGeometry(const sdf::Geometry &_geometry, pxr::UsdStageRefPtr &_stage,
+      const std::string &_path)
+  {
+    bool typeParsed = false;
+    switch (_geometry.Type())
+    {
+      case sdf::GeometryType::BOX:
+        typeParsed = ParseSdfBoxGeometry(_geometry, _stage, _path);
+        break;
+      case sdf::GeometryType::CYLINDER:
+        typeParsed = ParseSdfCylinderGeometry(_geometry, _stage, _path);
+        break;
+      case sdf::GeometryType::SPHERE:
+        typeParsed = ParseSdfSphereGeometry(_geometry, _stage, _path);
+        break;
+      case sdf::GeometryType::MESH:
+        typeParsed = ParseSdfMeshGeometry(_geometry, _stage, _path);
+        break;
+      case sdf::GeometryType::CAPSULE:
+        typeParsed = ParseSdfCapsuleGeometry(_geometry, _stage, _path);
+        break;
+      case sdf::GeometryType::PLANE:
+        typeParsed = ParseSdfPlaneGeometry(_geometry, _stage, _path);
+        break;
+      case sdf::GeometryType::ELLIPSOID:
+      case sdf::GeometryType::HEIGHTMAP:
+      default:
+        std::cerr << "Geometry type is either invalid or not supported\n";
+    }
+
+    // Set the collision. In SDF, the collision is defined separately from
+    // the visual's geometry (in case a user wants to define a simpler collision,
+    // for example). In USD, the collision can be attached to the geometry so
+    // that the collision shape is the geometry shape.
+    // TODO(adlarkin) support the option of a different collision shape,
+    // especially for meshes - here's more information about how to do this:
+    // https://graphics.pixar.com/usd/release/wp_rigid_body_physics.html?highlight=collision#turning-meshes-into-shapes
+    if (typeParsed)
+    {
+      auto geomPrim = _stage->GetPrimAtPath(pxr::SdfPath(_path));
+      if (!geomPrim)
+      {
+        std::cerr << "Internal error: unable to get prim at path ["
+                  << _path << "], but a geom prim should exist at this path\n";
+        return false;
+      }
+
+      if (!pxr::UsdPhysicsCollisionAPI::Apply(geomPrim))
```

Comment:
you should create two schemas, one for the visual and another one for the collision, even if this implies to duplicate the mesh.

Reply:
I'm not sure that I understand what you mean - are you saying that the geometry should have two schemas applied to it (i.e., collision and something else)? What is the visual schema you mentioned, can you share a link to documentation/API somewhere?